Fix boundary issue in `pushFrequency`

mkmoisen opened the pull request.

nieldlr commented:
Hi @mkmoisen,

sorry I missed this PR. Thanks for this!

Could you perhaps explain what this change dpes? Would be happy to merge this. I just want to understand.

Cheers!

mkmoisen commented:
Hi @nieldlr , 

For example, if `mean = 10` and `sd = 5`, the following values would be calculated [here](https://github.com/nieldlr/hanzi/blob/master/lib/dictionary.js#L209):
    
    lowrange = 5  // mean - sd
    midrange = [15, 5] // [mean + sd, lowrange]
    highrange = 15  // mean + sd

If a given `wordfreq[word.simplified]` just happened to be exactly 5 or 15 (this might be really rare) , the current code base would not include it in either `lowfreq`, `midfreq`, or `highfreq`, since it only uses `<` or `>` comparisons:

    if (wordfreq[word.simplified] < lowrange) { 
            lowfreq.push(word);
          }
          if (
            wordfreq[word.simplified] > midrange[1] &&
            wordfreq[word.simplified] < midrange[0]
          ) {
            midfreq.push(word);
          }
          if (wordfreq[word.simplified] > highrange) {
            highfreq.push(word);
    }

The PR ensures that a given `wordfreq[word.simplified]` of 5  would be put in `midfreq` and a `wordfreq[word.simplified]` of 15 would go into highfreq:

      if (wordfreq[word.simplified] < lowrange) {
            lowfreq.push(word);
          }
          if (
            wordfreq[word.simplified] >= midrange[1] &&
            wordfreq[word.simplified] < midrange[0]
          ) {
            midfreq.push(word);
          }
          if (wordfreq[word.simplified] >= highrange) {
            highfreq.push(word);
          }
    }

nieldlr commented:
Heya @mkmoisen,

this makes total sense.  Great catch on this. Merging this now!
